src/scales/scale.time.js

```diff
@@ -9,28 +9,32 @@ import {_lookup, _lookupByKey} from '../helpers/helpers.collection';
  * @typedef { import("../core/core.adapters").Unit } Unit
  */
 
+/**
+ * @typedef {{common: boolean, size: number, steps?: number}} Interval
+ */
+
 // Integer constants are from the ES6 spec.
 const MAX_INTEGER = Number.MAX_SAFE_INTEGER || 9007199254740991;
 
 /**
- * @type {Map<Unit, {common: boolean, size: number, steps?: number}>}
+ * @type {Object<Unit, Interval>}
  */
-const INTERVALS = new Map();
-INTERVALS.set('millisecond', {common: true, size: 1, steps: 1000});
-INTERVALS.set('second', {common: true, size: 1000, steps: 60});
-INTERVALS.set('minute', {common: true, size: 60000, steps: 60});
-INTERVALS.set('hour', {common: true, size: 3600000, steps: 24});
-INTERVALS.set('day', {common: true, size: 86400000, steps: 30});
-INTERVALS.set('week', {common: false, size: 604800000, steps: 4});
-INTERVALS.set('month', {common: true, size: 2.628e9, steps: 12});
-INTERVALS.set('quarter', {common: false, size: 7.884e9, steps: 4});
-INTERVALS.set('year', {common: true, size: 3.154e10});
+const INTERVALS = {
+	millisecond: {common: true, size: 1, steps: 1000},
+	second: {common: true, size: 1000, steps: 60},
+	minute: {common: true, size: 60000, steps: 60},
+	hour: {common: true, size: 3600000, steps: 24},
+	day: {common: true, size: 86400000, steps: 30},
+	week: {common: false, size: 604800000, steps: 4},
+	month: {common: true, size: 2.628e9, steps: 12},
+	quarter: {common: false, size: 7.884e9, steps: 4},
+	year: {common: true, size: 3.154e10}
+};
 
 /**
  * @type {Unit[]}
  */
-const UNITS = [];
-INTERVALS.forEach((v, k) => UNITS.push(k));
+const UNITS = /** @type Unit[] */(Object.keys(INTERVALS));
```

Comment:
Kinda surprised the type doesn't get inferred here

Reply:
Yeah, it's a bit annoying, but was a decision TypeScript made explicitly: https://github.com/microsoft/TypeScript/pull/12253#issuecomment-263132208

Reply:
I definitely agree with the reasoning in that comment. I wonder if they would return the correct type if the object was frozen with `Object.freeze`

Reply:
That doesn't work now. I'm not quite sure they could implement that because then the return type of `keys` would have to be one of two different types depending on whether the object was frozen or not. The compiler wouldn't be able to tell which it was, so the type here would then be `string[]|Unit[]`, which still isn't what we want